src/django_components/slots.py

```diff
@@ -498,3 +512,20 @@ def _report_slot_errors(
         if fuzzy_slot_name_matches:
             msg += f"\nDid you mean '{fuzzy_slot_name_matches[0]}'?"
         raise TemplateSyntaxError(msg)
+
+
+name_escape_re = re.compile(r"[^\w]")
+def _escape_slot_name(name: str) -> str:
```

Comment:
This function implements a graceful slot name resolution, as described in https://github.com/EmilStenstrom/django-components/issues/280#issuecomment-2081180940, so that users may define slots with non-identifier characters like "my slot!", but it can still be access via `component_vars.is_filled.my_slot_`